Mathlib/Data/Nat/Factorial/SuperFactorial.lean

```diff
@@ -79,6 +84,100 @@ theorem det_vandermonde_id_eq_superFactorial (n : ℕ) :
+
+theorem eval_matrixOfPolynomials_eq_vandermonde_mul_matrixOfPolynomials {n : ℕ}
+    (v : Fin n → R) (p : Fin n → R[X]) (h_deg : ∀ i, (p i).natDegree = i) :
+    Matrix.of (fun i j => ((p j).eval (v i))) =
+    (Matrix.vandermonde v) * (Matrix.of (fun (i j : Fin n) => (p j).coeff i)) := by
+  ext i j
+  rw [Matrix.mul_apply, Polynomial.eval, Matrix.of_apply, Polynomial.eval₂]
+  simp only [eq_intCast, Int.cast_id, Matrix.vandermonde]
+  have : (p j).support ⊆ range n := supp_subset_range <| h_deg j ▸ Fin.prop j
+  rw [sum_eq_of_subset _ (fun j => zero_mul ((v i) ^ j)) this, ← Fin.sum_univ_eq_sum_range]
+  congr
+  ext k
+  rw [mul_comm, Matrix.of_apply, RingHom.id_apply]
+
+theorem matrixOfPolynomials_blockTriangular {n : ℕ} (p : Fin n → R[X])
+    (h_deg : ∀ i, (p i).natDegree  = i) :
+    Matrix.BlockTriangular (Matrix.of (fun (i j : Fin n) => (p j).coeff i)) id :=
+  fun _ j h => coeff_eq_zero_of_natDegree_lt (h_deg j ▸ h)
+
+theorem det_matrixOfPolynomials {n : ℕ} (p : Fin n → R[X])
+    (h_deg : ∀ i, (p i).natDegree  = i) (h_monic : ∀ i,  Monic <| p i) :
+    (Matrix.of (fun (i j : Fin n) => (p j).coeff i)).det = 1 := by
+  rw [Matrix.det_of_upperTriangular (matrixOfPolynomials_blockTriangular p h_deg)]
+  convert prod_const_one
+  rw [Matrix.of_apply]
+  rename_i x _
+  convert h_monic x
+  unfold Monic leadingCoeff
+  rw [h_deg x]
+
+theorem vandermonde_det_eq_eval_matrixOfPolynomials_det {n : ℕ} (v : Fin n → R) (p : Fin n → R[X])
+    (h_deg : ∀ i, (p i).natDegree  = i) (h_monic : ∀ i,  Monic <| p i) :
+    (Matrix.vandermonde v).det = (Matrix.of (fun i j => ((p j).eval (v i)))).det := by
+  rw [eval_matrixOfPolynomials_eq_vandermonde_mul_matrixOfPolynomials v p h_deg, Matrix.det_mul,
+      det_matrixOfPolynomials p h_deg h_monic, mul_one]
+
+theorem eval_matrixOfPolynomials_eq_mul_matrix_of_choose {n : ℕ} (v : Fin n → ℕ) :
+    (Matrix.of (fun (i j : Fin n) =>
+    ((fun n => descPochhammer ℤ n) j).eval ((fun k => (v k : ℤ)) i)) ).det =
+    (∏ i : Fin n,  Nat.factorial i) *
+    (Matrix.of (fun (i j : Fin n)  => ((Nat.choose ((v i)) (j : ℕ)) : ℤ))).det := by
+  convert Matrix.det_mul_row (fun (i : Fin n)  => ((Nat.factorial (i : ℕ)):ℤ)) _
+  · rw [Matrix.of_apply, descPochhammer_int_eq_descFactorial _ _]
+    congr
+    exact Nat.descFactorial_eq_factorial_mul_choose _ _
+  · rw [Nat.cast_prod]
+
+theorem superFactorial_dvd_vandermonde_det {n : ℕ} (v : Fin (n + 1) → ℤ) :
+    ↑(Nat.superFactorial n) ∣ (Matrix.vandermonde v).det := by
+  let m' := List.minimum ((univ : Finset (Fin (n + 1))).image v).toList
+  have : ((univ : Finset (Fin (n + 1))).image v).toList ≠ List.nil := by
+    simp only [ne_eq, toList_eq_nil, image_eq_empty]
+    refine Nonempty.ne_empty univ_nonempty
+  cases' Option.isSome_iff_exists.mp <| Option.ne_none_iff_isSome.mp <|
+      List.minimum_ne_top_of_ne_nil this with m hm
+  have h_min : ∀  l, l ∈ ((univ : Finset (Fin (n + 1))).image v).toList →  m' ≤ l := by
+     exact fun l a ↦ List.minimum_le_of_mem' a
```

Comment:
We don't need to take the minimum of a list since `Finset.inf'` does basically the same thing:
```suggestion
  let m := inf' univ ⟨0, mem_univ _⟩ v
```